sklearn/datasets/_arff_parser.py

```diff
@@ -204,7 +204,10 @@ def _io_to_generator(gzip_file):
         if len(dfs) >= 2:
             dfs[0] = dfs[0].astype(dfs[1].dtypes)
 
-        frame = pd.concat(dfs, ignore_index=True)
+        # liac-arff parser does not depend on NumPy and uses None to represent
+        # missing values. To be consistent with the pandas parser, we replace
+        # None with np.nan.
+        frame = pd.concat(dfs, ignore_index=True).fillna(value=np.nan)
```

Comment:
An alternative here (but more invasive) would be to use `.convert_dtypes` and make type inferences to use `pd.NA` as a missing marker.

We could introduce this step in both parsers. However, as stated in the pandas documentation, if `pd.NA` as additional support for new dtypes, then we could get a change of behaviour without further notice.

It might be just safer to keep `np.nan` as the marker for the moment.